biopandas/mmcif/mmcif_parser.py

```diff
@@ -0,0 +1,444 @@
+# The CIF parser relieves heavily on the parser from PDBJapan (author: Gert-Jan Bekker).
+# Python CIF parser: https://gitlab.com/pdbjapan/tools/cif-parsers
+# License: MIT
+#   See https://gitlab.com/pdbjapan/tools/cif-parsers/blob/master/LICENSE
+import gzip
+import re
+
+try:
+    import json
```

Comment:
Shouldn't 

    import json

work on all modern Python versions? I think you maybe added it because you were concerned that it doesn't work in old Python versions. However, I think people who are using older versions have bigger issues due to the type hints in this code :P. So, I think it is safe to assume that this import here is not an issue and the code can be simplified to just

    import json

Reply:
So, the mmCIF parser implementation comes from PDBJapan. I would think supporting older python versions is their goal too. 

Now, in terms of how we proceed, I think it's your call as the maintainer. AFAIK numpy and pandas have [dropped](https://numpy.org/news/) offical [support](https://pandas.pydata.org/docs/getting_started/install.html) for <3.8 so we could take that as indication to let go of the past 😁 . Do you know if there are a lot of legacy projects that depend on biopandas? Hopefully they'd have pinned a version if so.

Happy to remove the ugly old python support if that's what you'd prefer.

Reply:
I am honestly fine with not supporting Python 2.7 anymore. I think that ship has sailed :P. The reason why I brought this up though is that we have type hints in your code, which are only supported from Python 3.5 onwards anyway. So in this case, code wouldn't be compatible to older Python versions anyways.

Reply:
PS: I am also fine with not worrying about pre Python 3.8. To be honest, maintaining a library is already a lot of work for a current Python version :)

Reply:
Absolutely! I'd be very happy if we could keep support for 3.7 for a while at least. My own project, [Graphein](https://github.com/a-r-j/graphein) depends quite heavily on biopandas and I've had explicit requests to support 3.7. I don't think there's anything here that would prevent that though.

Reply:
Of course, I am happy to not break Python 3.7 support. I do think there is nothing really that should prevent it though. When I have time this weekend, I will try to also shift this project here to GitHub actions and add Python 3.7 as minimum requirement. Graphein looks like a cool library btw. Super impressive.